### PR TITLE
src/Makefile: Link libatomic on POWER systems

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This ensures that fallbacks for unsupported atomic operations are available for POWER systems.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -150,6 +150,11 @@ DEBUG=-g -ggdb
 # Linux ARM32 needs -latomic at linking time
 ifneq (,$(findstring armv,$(uname_M)))
         FINAL_LIBS+=-latomic
+else
+# Linux POWER needs -latomic at linking time
+ifneq (,$(findstring ppc,$(uname_M)))
+        FINAL_LIBS+=-latomic
+endif
 endif
```
Comment (PingXie) [inline src/Makefile:158]: can we combine the two?

```suggestion
ifeq (,$(findstring $(uname_M),armv ppc)) 
    FINAL_LIBS+=-latomic
endif
```

Comment (Conan-Kudo) [reply src/Makefile:158]: This should be addressed now.

**Comment by Conan-Kudo** [comment]
Looks like this version broke macOS and the jemalloc version ☹️

**Comment by Conan-Kudo** [comment]
I've rolled back to my original one for now, since that works as intended.

**Review by PingXie — APPROVED** [review]

**Pull request merged**